specification/Metadata/README.md

```diff
@@ -0,0 +1,446 @@
+
+### Version
+
+Schema version (`version`) is an application-specific identifier for a given schema revision. Version must be a string, and should be syntactically compatible with [SemVer](https://semver.org/).
+
+> **Example:** Valid semantic versions include strings like `0.1.2`, `1.2.3`, and `1.2.3-alpha`.
+
+### Enums
+
+An enum consists of a set of named values, represented as `(string, integer)` pairs. The following enum value types are supported: `INT8`, `UINT8`, `INT16`, `UINT16`, `INT32`, `UINT32`, `INT64`, and `UINT64`. See the [Type](#type) section for definitions of each. Smaller enum types limit the range of possible enum values, and allow more efficient binary encoding. For unsigned value types, enum values most be non-negative. Duplicate names or values within the same enum are not allowed.
+
+The example below defines a "species" enum with three possible tree species, as well as an "Unknown" value.
+
+| name      | value |
+|-----------|------:|
+| "Oak"     |     0 |
+| "Pine"    |     1 |
+| "Maple"   |     2 |
+| "Unknown" |    -1 |
+
+### Classes
+
+Classes represent categories of similar entities, and are defined by a collection of one or more properties shared by the entities of a class. Each class has a unique ID within the schema, and each property has a unique ID within the class, to be used for references within the schema and externally.
+
+### Properties
+
+#### Overview
+
+Properties describe the type and structure of values that may be associated with entities of a class. Entities may omit values for a property, unless the property is required. Entities must not contain values other than those defined by the properties of their class.
+
+> **Example:** The following example shows the basics of how classes describe the types of metadata. A `building` class describes the heights of various buildings in a dataset. Likewise, the `tree` class describes trees that have a height, species, and leaf color.
+>
+> **building**
+>
+> | property | componentType | required | noData |
+> |:---------|:--------------|:---------|:-------|
+> | height   | "FLOAT32"     | ✓        |        |
+>
+> **tree**
+>
+> | property  | componentType | required | noData    |
+> |:----------|:--------------|:---------|:----------|
+> | height    | "FLOAT32"     | ✓        |           |
+> | species   | "STRING"      |          | "Unknown" |
+> | leafColor | "STRING"      | ✓        |           |
+
+#### ID
+
+IDs (`id`) uniquely identify a property within a class, and must contain only alphanumeric characters and underscores. IDs should be human-readable (wherever possible) and camel-case. When IDs subject to these restrictions are not sufficiently clear for human readers, applications should also provide a property *name*.
```

Comment:
```suggestion
IDs (`id`) uniquely identify a property within a class, and must contain only alphanumeric characters and underscores to be machine-parseable. IDs should be camel case strings that are human-readable (wherever possible). When IDs subject to these restrictions are not sufficiently clear for human readers, applications should also provide a property *name*.
```

Reply:
I'm not sure what "machine-parseable" means here, or why non-alphanumeric characters aren't. We probably don't want to get into "should be valid variable names in [language]" here, would it be OK to include just the 2nd change?

Reply:
that's fair, just do the second change